lib/p2p/Peer.ts

```diff
@@ -239,6 +245,10 @@ class Peer extends EventEmitter {
     }
 
     return new Promise((resolve, reject) => {
+      const startTime = Date.now();
+      let retryDelay: number | undefined;
```

Comment:
How about we initialize `retryDelay` by setting it to `CONNECTION_RETRIES_MIN_DELAY`, and then move the line that doubles the retry delay to after the `retryDelay > Peer.CONNECTION_RETIES_MAX_PERIOD` check. It feels a bit cleaner that way, and saves a bunch of null checks on `retryDelay`.

Reply:
What do you mean it forced you to update the next value?

Reply:
I changed it, so you can look now. Nothing serious.